src/domains/ecommerce-common/03_cosmosdb.tf

```
@@ -144,6 +144,10 @@ locals {
         {
           keys   = ["email.data"]
           unique = false
+        },
+        {
+          keys   = ["userId"]
+          unique = false
```

userId is an optional field that may be missing in the scenario of a guest flow. Please fix this index to skip indexing when the field is missing.

```suggestion
          unique = false,
          condition = "exists(userId)"  # Add this condition to handle missing value
```

---

By default Text index is a Sparse index [MongoDb doc](https://www.mongodb.com/docs/manual/core/indexes/index-types/index-text/#sparse-property). This means that by default possible null values are already handled.

---

great!